server/src/main/java/org/apache/cassandra/sidecar/accesscontrol/authentication/MutualTlsAuthenticationHandler.java

```diff
@@ -0,0 +1,85 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.auth.authentication.CertificateCredentials;
+import io.vertx.ext.auth.authentication.CredentialValidationException;
+import io.vertx.ext.auth.mtls.MutualTlsAuthentication;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.HttpException;
+import io.vertx.ext.web.handler.impl.AuthenticationHandlerImpl;
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+
+/**
+ * Handler for verifying user certificates for Mutual TLS authentication. {@link MutualTlsAuthenticationHandler} can be
+ * chained with other {@link io.vertx.ext.web.handler.AuthenticationHandler} implementations.
+ */
+public class MutualTlsAuthenticationHandler extends AuthenticationHandlerImpl<MutualTlsAuthentication>
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(MutualTlsAuthenticationHandler.class);
+    private final ExecutorPools executorPools;
+
+    public MutualTlsAuthenticationHandler(MutualTlsAuthentication authProvider,
+                                          ExecutorPools executorPools)
+    {
+        super(authProvider);
+        this.executorPools = executorPools;
+    }
+
+    @Override
+    public void authenticate(RoutingContext ctx, Handler<AsyncResult<User>> handler)
+    {
+        if (!ctx.request().isSSL())
+        {
+            ctx.response().setStatusCode(HttpResponseStatus.BAD_REQUEST.code()).end();
+            return;
+        }
+
+        CertificateCredentials certificateCredentials = CertificateCredentials.fromHttpRequest(ctx.request());
+        executorPools
+        .service()
+        .executeBlocking(promise -> {
+            authProvider.authenticate(certificateCredentials)
+                        .onSuccess(promise::complete)
+                        .onFailure(cause -> {
+                            if (cause instanceof CredentialValidationException)
+                            {
+                                // If credentials are invalid, reject the promise with a BAD_REQUEST error
+                                promise.fail(new HttpException(HttpResponseStatus.BAD_REQUEST.code(),
+                                                               "Error validating credentials passed"));
+                            }
+                            else
+                            {
+                                // In case of other failures, reject the promise with an UNAUTHORIZED error
+                                promise.fail(new HttpException(HttpResponseStatus.UNAUTHORIZED.code()));
+                            }
+                        });
+        })
+        .onSuccess(result -> handler.handle(Future.succeededFuture((User) result)))
+        .onFailure(cause -> handler.handle(Future.failedFuture(cause)));
```

Comment:
```suggestion
        authProvider.authenticate(certificateCredentials)
                    .onSuccess(result -> handler.handle(Future.succeededFuture(result)))
                    .onFailure(cause -> ctx.fail(new HttpException(HttpResponseStatus.UNAUTHORIZED.code(), cause)));
```

1. `authProvider.authenticate` is already async. No need to dispatch to another executor pool again. 
2. All exceptions should be mapped to `UNAUTHORIZED`, IMO. Invalid cert is not a bad request. 
3. Do not call the next handler when auth failed. Simply fail the request instead.

Reply:
The reason we hand off to executor pool, is because, we use `identityRoleCache`, during loads, since database calls are involved, it blocks thread. I can move hand off to executor pool inside `CassandraIdentityExtractor`

Reply:
Regarding 3. since we use `ChainAuthHandler` calling next, allows the next authentication handler to be called. `ChainAuthHandler` finally returns `UNAUTHORIZED` if all handlers fail.